Tests/UnitTests/Attribution/AttributionPosterTests.swift

```diff
@@ -57,245 +55,118 @@ class AttributionPosterTests: TestCase {
                                                    attributionFetcher: self.attributionFetcher,
                                                    subscriberAttributesManager: self.subscriberAttributesManager)
         self.resetAttributionStaticProperties()
-        self.backend.stubbedPostAttributionDataCompletionResult = (nil, ())
+        self.backend.stubbedPostAdServicesTokenCompletionResult = .success(())
     }
 
+    override func tearDown() {
+            UserDefaults.standard.removePersistentDomain(forName: userDefaultsSuiteName)
+            UserDefaults.standard.synchronize()
+            resetAttributionStaticProperties()
+            super.tearDown()
+        }
+
     private func resetAttributionStaticProperties() {
         if #available(iOS 14, macOS 11, tvOS 14, *) {
             MockTrackingManagerProxy.mockAuthorizationStatus = .authorized
         }
 
-        MockAttributionTypeFactory.shouldReturnAdClientProxy = true
         MockAttributionTypeFactory.shouldReturnTrackingManagerProxy = true
-        MockAdClientProxy.requestAttributionDetailsCallCount = 0
     }
 
-    override func tearDown() {
-        super.tearDown()
-        UserDefaults.standard.removePersistentDomain(forName: userDefaultsSuiteName)
-        UserDefaults.standard.synchronize()
-        resetAttributionStaticProperties()
+}
+
+#if canImport(AdServices)
+@available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
+class AdServicesAttributionPosterTests: BaseAttributionPosterTests {
```

Comment:
Ditto this needs an availability check in `setUpWithError`.

Reply:
doing the availability check in `setUpWithError` leads to the "redundant availability check" error... and it seems like we've avoided making more specific checks in the `AvailabilityChecks` class?

Reply:
I think without it these tests will fail on iOS 13 though (which won't be caught until this is merged in `main`.